gammapy/image/models/new.py

```diff
@@ -0,0 +1,82 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Morphological models for astrophysical gamma-ray sources - new implementation
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from ...utils.modeling import Parameter, ParameterList
+from astropy.coordinates.angle_utilities import angular_separation
+from astropy.coordinates import Angle, Longitude, Latitude
+import numpy as np
+import astropy.units as u
+
+__all__ = [
+    'SkySpatialModel',
+    'SkyGaussian2D',
+]
+
+
+class SkySpatialModel(object):
+    """SkySpatial model base class.
+    """
+
+    def __str__(self):
+        ss = self.__class__.__name__
+        ss += '\n\nParameters: \n\n\t'
+
+        table = self.parameters.to_table()
+        ss += '\n\t'.join(table.pformat())
+
+        if self.parameters.covariance is not None:
+            ss += '\n\nCovariance: \n\n\t'
+            covar = self.parameters.covariance_to_table()
+            ss += '\n\t'.join(covar.pformat())
+        return ss
+
+    def __call__(self, lon, lat):
+        """Call evaluate method of derived classes"""
+        kwargs = dict()
+        for par in self.parameters.parameters:
+            kwargs[par.name] = par.quantity
+
+        return self.evaluate(lon, lat, **kwargs)
+
+    def copy(self):
+        """A deep copy."""
+        return copy.deepcopy(self)
+
+
+class SkyGaussian2D(SkySpatialModel):
+    r"""Two-dimensional symmetric Gaussian model.
+
+    .. math::
+
+        \phi(x, y) = \frac{1}{2\pi\sigma^2} \exp{\left(-\frac{1}{2}
+            \frac{\theta^2}{\sigma^2}\right)}
+
+    where :math:`\theta` is the sky separation
+
+    Parameters
+    ----------
+    lon_mean : `~astropy.coordiantes.Longitude`
+        :math:`x_0`
+    lat_mean : `~astropy.coordinates.Latitude`
+        :math:`y_0`
+    sigma : `~astropy.coordinates.Angle`
+        :math:`\sigma`
+    """
+
+    def __init__(self, lon_mean, lat_mean, sigma):
+        self.parameters = ParameterList([
+            Parameter('lon_mean', Longitude(lon_mean)),
+            Parameter('lat_mean', Latitude(lat_mean)),
+            Parameter('sigma', Angle(sigma))
+        ])
+
+    @staticmethod
+    def evaluate(lon, lat, lon_mean, lat_mean, sigma):
```

Comment:
I'm not really sure how we should handle the sky coordinates for model evaluation. I do not like the fact that we are bound to Galactic coordinates here, but it is probably the easiest solution. Alternatively one could define the coordinate system during initialization of the model by introducing a `frame` attribute (similar to `SkyCoord`) and store it as class attribute.  One could then think about accepting a `SkyCoord` object in the `__call__()` method:

```
model = SkyGaussian2D(x=0*u.deg, y=0*u.deg, frame='galactic')
image = SkyImage()
values = model(image.coordinates())
```

Alternatively keep as is and later introduce a convenience property `SpatialModel.position`, which returns a `SkyCoord` object.

Reply:
The basic idea is to only use Numpy or maybe Quantity like here in model evaluation, and to use SkyCoord, Time, ... not in the model evaluate / likelihood fit. We could add convenience properties though, e.g. a `skycoord` to get and set center of the spatial models.
But then again, the model would probably also have to do frame handling, ... let's continue that discussion after we have a first working 3D fit and a few results.
(Gammapy CTA presentation next week Thursday morning)